test/spec/services/angular-apimock.js

```diff
@@ -409,6 +412,83 @@ describe('Service: apiMock', function () {
 
 			});
 
+			describe('allow regexp for apiPath option', function () {
+
+				beforeEach(function() {
+					// apiMockProvider.config({apiPath: [/\/(aPI)/i]});
+					apiMockProvider.config({apiPath: /\/(aPi|UPI|APU)/i});
+					setGlobalCommand(true);
+				});
+
+				afterEach(function() {
+					apiMockProvider.config({apiPath: '/api'});
+					unsetGlobalCommand();
+				});
+
+				it('should redirect when match', function () {
+					defaultRequest.url = '/api/pokemon';
+					defaultExpectPath = '/mock_data/pokemon.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should redirect any match', function () {
+					defaultRequest.url = '/UPI/pokemon';
+					defaultExpectPath = '/mock_data/pokemon.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should NOT redirect not matched', function () {
+					defaultRequest.url = '/EPI/picachu';
+					defaultExpectPath = '/EPI/picachu';
+					expectHttpFailure();
+				});
+
+			});
+
+			describe('allow strings array for apiPath option', function () {
+
+				beforeEach(function() {
+					apiMockProvider.config({apiPath: ['/other/api', '/api', /\/v(2|3|4)api/i]});
```

Comment:
Totally agree and great idea, will do and let you know when pushed to my form for you to review.

Thanks for all the feedback, appreciated.

BTW, I was looking around you 1.5K size but that would difficult as it was almos 4K when starting. :+1: